### DISCUSSION TITLE
validation does not work on the "onChange" event

### DISCUSSION THREAD

**Discussion by lordenas** [react-hook-form/react-hook-form#7783] [Q&A] [answered]
I want to make the validation in the "Repeat password" field work every time the "Password" field changes, but I can't figure out how I can achieve this behavior.

I provided an example where I set mode: "onChange", but when editing the first field, the second field's validation will only work if itSubmits or when editing "Repeat password"

https://codesandbox.io/s/react-hook-form-password-match-check-standard-validation-forked-lo3w8?file=/src/index.js

In my case, things are much more complicated, I use yupResolver with validationSchema

**Comment by Moshyfawn** [comment, marked as answer]
You can use the `deps` option to provide a list of field names to validate `onChange`.
```js
{...register("password", { deps: ["repeatPassword"] })}
```

**Reply by lordenas** [reply]
tell me how to do it correctly if you use Controller?

**Reply by Moshyfawn** [reply]
Pass down the same `deps` list into the `rules` prop on `Controller`
```js
<Controller rules={{ deps: ["repeatPassword"] }} />
```

**Reply by lordenas** [reply]
yes, that's what you need. Thank you!
